db/migrate/20170626141302_drop_public_body_translated_columns.rb

```diff
@@ -0,0 +1,54 @@
+# -*- encoding : utf-8 -*-
+class DropPublicBodyTranslatedColumns < ActiveRecord::Migration
+  def up
+    remove_column :public_bodies, :name
+    remove_column :public_bodies, :short_name
+    remove_column :public_bodies, :request_email
+    remove_column :public_bodies, :url_name
+    remove_column :public_bodies, :notes
+    remove_column :public_bodies, :first_letter
+    remove_column :public_bodies, :publication_scheme
+  end
+
+  def down
```

Comment:
Okay here's what I think we should do:

- Add a data migration in the `up` migration, creating a default locale translation _only if_ one doesn't already exist. This should be a no-op but it covers any edge cases that might have happened.
- If we can't find a translation for the default locale (shouldn't be a problem, given the above) fall back to the first translation we can find so that the public body constraints are met.

```diff
diff --git a/db/migrate/20170626141302_drop_public_body_translated_columns.rb b/db/migrate/20170626141302_drop_public_body_translated_columns.rb
index 060cc54..006973b 100644
--- a/db/migrate/20170626141302_drop_public_body_translated_columns.rb
+++ b/db/migrate/20170626141302_drop_public_body_translated_columns.rb
@@ -1,6 +1,26 @@
 # -*- encoding : utf-8 -*-
 class DropPublicBodyTranslatedColumns < ActiveRecord::Migration
   def up
+    PublicBody.transaction do
+      PublicBody.find_each do |record|
+        translation = record.translation_for(I18n.default_locale) ||
+          record.translations.build(:locale => I18n.default_locale)
+
+        if translation.new_record?
+          id = record.id
+          puts "Creating default locale translation for public body #{ id }"
+
+          fields = record.translated_attributes
+          fields.each do |attribute_name, attribute_type|
+            translation[attribute_name] =
+              record.read_attribute(attribute_name, :translated => false)
+          end
+
+          translation.save!
+        end
+      end
+    end
+
     remove_column :public_bodies, :name
     remove_column :public_bodies, :short_name
     remove_column :public_bodies, :request_email
@@ -22,19 +42,25 @@ class DropPublicBodyTranslatedColumns < ActiveRecord::Migration
     # Migrate the data back - wrapped in its own transaction so the data will
     # be updated before the constraints are reapplied
     PublicBody.transaction do
-      PublicBody.find_each do |pb|
-        translated = pb.translation_for(I18n.default_locale)
+      PublicBody.find_each do |record|
+        translated = record.translation_for(I18n.default_locale)
+        translated = record.translations.first unless translated.persisted?
+
+        if translated.new_record? || translated.nil?
+          puts "No translations for public body #{record.id}"
+        end
 
         # Create a hash containing the translated column names and their values
-        pb.translated_attribute_names.inject(fields_to_update={}) do |f, name|
+        record.translated_attribute_names.inject(fields_to_update={}) do |f, name|
           f.update({name.to_sym => translated[name.to_s]})
         end
 
         # Now, update the actual model's record with the hash (using the
         # ActiveRecord::Relation method update_all to force the use of an
-        # UPDATE statement rather than pb.update_atrributes which will
+        # UPDATE statement rather than record.update_atrributes which will
         # use the overridden attribute setters and update translations instead)
-        PublicBody.where(:id => pb.id).update_all(fields_to_update)
+        puts "Migrating default locale translation to public body #{record.id}"
+        PublicBody.where(:id => record.id).update_all(fields_to_update)
       end
     end
```